usr/share/rear/layout/save/GNU/Linux/210_raid_layout.sh

```diff
@@ -15,14 +16,20 @@ if [ -e /proc/mdstat ] &&  grep -q blocks /proc/mdstat ; then
             fi
 
             # We use the detailed mdadm output quite a lot
-            mdadm --misc --detail $device > $TMP_DIR/mdraid
+            tmpfile="$TMP_DIR/mdraid.$name"
+            mdadm --misc --detail $device > $tmpfile
+            ( echo "mdadm --misc --detail $device" ; cat $tmpfile ) | sed -e 's/^/# /'
 
             # Gather information
-            metadata=$( grep "Version" $TMP_DIR/mdraid | tr -d " " | cut -d ":" -f "2")
-            level=$( grep "Raid Level" $TMP_DIR/mdraid | tr -d " " | cut -d ":" -f "2")
-            uuid=$( grep "UUID" $TMP_DIR/mdraid | tr -d " " | cut -d "(" -f "1" | cut -d ":" -f "2-")
-            layout=$( grep "Layout" $TMP_DIR/mdraid | tr -d " " | cut -d ":" -f "2")
-            chunksize=$( grep "Chunk Size" $TMP_DIR/mdraid | tr -d " " | cut -d ":" -f "2" | sed -r 's/^([0-9]+).+/\1/')
+            metadata=$( grep "Version" $tmpfile | tr -d " " | cut -d ":" -f "2")
+            level=$( grep "Raid Level" $tmpfile | tr -d " " | cut -d ":" -f "2")
+            uuid=$( grep "UUID" $tmpfile | tr -d " " | cut -d "(" -f "1" | cut -d ":" -f "2-")
```

Comment:
I was wondering about the overcomplicated looking code
to extract the UUID, here is the reason:
https://github.com/rear/rear/commit/b9011f1764c3d4e6ce547c4ede88e32e6fe8a389

I wished so much comments in the code that explain
why strange looking code is as is - I spend so much time
with needless reverse engineering why things are as is :-(

Reply:
@jsmeix or awk: code such as `| awk '{print $...}'` is being used already.

Reply:
I am not a regular awk user.
I am basically an awk script kiddie.
For example I already fail to understand what
https://www.gnu.org/software/gawk/manual/html_node/Default-Field-Splitting.html
means. It reads
```
Fields are normally separated by whitespace sequences
(spaces, TABs, and newlines), not by single spaces.
```
so it seems single spaces do not separate fields but in practice
```
# echo "foo bar baz" | awk '{print $2}'
bar
```
awk separates fields by single spaces - at least for me (sometimes?) ;-)